### PR TITLE
[BACK-2959] [BACK-2961] Add executors for patient, clinician and tag merge plans

### PR DESCRIPTION
**Pull request by toddkazakov**

### CODE REVIEW

**Review by lostlevels — APPROVED** [review]

File: clinics/merge/patients.go [hunk 1]
```diff
@@ -219,3 +231,153 @@ func getPatientTagNames(patient patients.Patient, tags map[string]*clinics.Patie
...
+func (p *PatientPlanExecutor) mergeTags(ctx context.Context, plan PatientPlan, target clinics.Clinic) error {
+	tagNames := map[string]struct{}{}
+	for _, name := range plan.PostMigrationTagNames {
+		tagNames[name] = struct{}{}
+	}
+	tagIds := make([]primitive.ObjectID, 0, len(tagNames))
+	for _, tag := range target.PatientTags {
+		if _, ok := tagNames[tag.Name]; ok {
+			tagIds = append(tagIds, *tag.Id)
+			delete(tagNames, tag.Name)
```
[the remark this answers is not in the document]
Comment (toddkazakov) [reply clinics/merge/patients.go:341]: Yes. `plan.PostMigrationTagNames` will always be unique, but during testing I saw few examples where we had a duplicate in `target.PatientTags`.

File: clinics/merge/clinicians.go [hunk 2]
```diff
@@ -56,6 +60,16 @@ func (c ClinicianPlans) PreventsMerge() bool {
 	return PlansPreventMerge(c)
 }
 
+func (c ClinicianPlans) PendingInvitesByWorkspace() map[string]int{
+	result := make(map[string]int)
+	for _, p := range c {
+		if p.IsPendingInvite() {
+			result[p.Workspaces[0]] = result[p.Workspaces[0]] + 1
```
Comment (lostlevels) [inline clinics/merge/clinicians.go:67]: Can't a `ClinicianPlan` contain more than one workspace? Should those be handled?

Comment (toddkazakov) [reply clinics/merge/clinicians.go:67]: A pending invite is bound to a single workspace

**Pull request merged**